### DISCUSSION TITLE
authWithOAuth2 - 404 Not Found Error

### DISCUSSION THREAD

**Discussion by maxwell-oroark** [pocketbase/pocketbase#2452] [Q&A] [answered]
I am attempting to use the new all-in-one oAuth2 sign in method and not sure if I am missing a step.

I've configured my strava client and shared the client ID and secret with via the pocketbase admin UI.

On the client-side I am invoking the function like so: ` pb.collection("users").authWithOAuth2({ provider: "strava" });`

Which seems to load my oAuth app fine:
<img width="1002" alt="Screen Shot 2023-05-09 at 10 20 35 AM" src="https://github.com/pocketbase/pocketbase/assets/6313131/c046d2d6-19d0-4a47-a430-2176a7e2f543">

However, after authorizing, the oAuth popup lands me at a 404 error route:
<img width="1001" alt="Screen Shot 2023-05-09 at 10 20 20 AM" src="https://github.com/pocketbase/pocketbase/assets/6313131/3d2464bf-33d4-4ca0-b285-c2f399d3cbda">

Am I missing another configuration step somewhere? I guess I am expecting my hosted pocketbase app to have live /api routes that are able to parse the payload from this authorization redirect and create a new strava user in my db. Is my understanding here correct?

Thanks in advance!

**Comment by ganigeorgiev** [comment, marked as answer]
What version of PocketBase do you use (_you can find it in the bottom right corner in the Admin UI_)?
Note that the all-in-one OAuth2 call (and the `/api/oauth2-redirect` route) was added in v0.15.0.

**Reply by maxwell-oroark** [reply]
Yes. That's exactly what my issue was. After upgrading my pocketbase to 15.3 it's working perfectly. Thanks alot
